content/docs/rfcs/43/README.md

```diff
@@ -0,0 +1,354 @@
+for enabling two devices to mutually authenticate 
+and securely exchange (arbitrary) information over the Waku network. 
+
+# Background / Rationale / Motivation
+
+In order to implement multi-device communications using one of the Noise session management mechanisms proposed in [37/WAKU2-NOISE-SESSIONS](https://rfc.vac.dev/spec/37/), 
+we require a protocol to securely exchange (cryptographic) information between 2 or more devices possessed by a user.
+
+Since, in this scenario, the devices would be close to each other, 
+authentication can be initialized by exchanging a QR code out-of-band 
+and then securely completed over the Waku network.
+
+The protocol we propose consists of two main subprotocols or *phases*:
+
+- [Device Pairing](#Device-Pairing): two phisically close devices initialize the *pairing* by exchanging a QR code out-of-band. The devices then exchange and authenticate their respective long-term device ID static key by exchanging handshake messages over the Waku network;
+- [Secure Transfer](#Secure-Transfer): the devices securely exchange information in encrypted form using key material obtained during a successful pairing phase. The communication will happen over the Waku network, hence the devices do not need to be phisically close in this phase.
+
+# Theory / Semantics
+
+## Device Pairing
+
+In the pairing phase, device `B` requests to be paired to a device `A`. 
+Once the two devices are paired, the devices will be mutually authenticated 
+and will share a Noise session within which they can securely exchange information.
+
+The request is made by exposing a QR code that, by default, has to be scanned by device `A`. 
+If device `A` doesn't have a camera while device `B` does, 
+[it is possible](#Rationale) to execute a slightly different pairing (with same security guarantees), 
+where `A` is exposing a QR code instead.
+
+This protocol is designed in order to achieve two main security objectives:
+
+- resistance to Man-in-the-Middle attacks;
+- provide network anonymity on devices' static keys, i.e. only paired devices will learn each other static key.
+
+### Employed Cryptographic Primitives
+
+- `H`: the underlying cryptographically-secure hash function, e.g. SHA-256;
+- `HKDF`: the key derivation function (based on `H`);
+- `Curve25519`: the underlying elliptic curve for Diffie-Hellman (DH) operations.
+
+### The `WakuPairing` Noise Handshake
+
+The devices execute a custom handshake derived from `XX`, 
+where they mutually exchange and authenticate their respective device static key 
+by exchanging messages over the content topic with the following [format](https://rfc.vac.dev/spec/23/#content-topic-format)
+
+```
+contentTopic = /{application-name}/{application-version}/wakunoise/1/sessions_shard-{shard-id}/proto
+```
+
+The handshake, detailed in next section, can be summarized as:
+
+```
+WakuPairing:
+a.   <- eB              {H(sB||r), contentTopicParams, messageNametag}
```

Comment:
Why `messageNameTag` is needed, it is explained in the RFC? 
Isn't the `contentTopic` enough already to filter messages belonging to the current communication? I assume `messageNameTag` is to reduce the number of tries and failures for decrypting messages not belonging to the current communication, right? In such a case, the more unique the `messageNameTag` the less anonymity is yielded. Would be good to explain it in the specs to let developers make a more informed decision on the trade-offs associated with the uniqueness of `messageNameTag`.